File: 4-1/임베실/week12/app/src/main/java/com/example/week12/MainActivity.java
package com.example.week12;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.NotificationCompat;
import androidx.core.app.NotificationManagerCompat;

import android.app.Notification;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.content.Context;
import android.os.Build;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {
    private final static String CHANNEL_ID = "channel_id";
    int channel_num;
    EditText title1;
    EditText content1;
    String title;
    String content;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button b1 = (Button) findViewById(R.id.Notify);
        Button b2 = (Button) findViewById(R.id.Update);
        Button b3 = (Button) findViewById(R.id.RemoveAll);



        channel_num = -1;

        createNoficationChannel();

        b1.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                channel_num  = channel_num + 1;
                title1 = (EditText) findViewById(R.id.title);
                content1 = (EditText) findViewById(R.id.contents);
                title = title1.getText().toString();
                content = content1.getText().toString();
                addNotification();
            }
        });

        b2.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                title1 = (EditText) findViewById(R.id.title);
                content1 = (EditText) findViewById(R.id.contents);
                title = title1.getText().toString();
                content = content1.getText().toString();
                addNotification();
                channel_num  = channel_num + 1;
            }
        });

        b3.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                removeNotification();
            }
        });

    }

    private void addNotification(){

        NotificationCompat.Builder builder = new NotificationCompat.Builder(this, CHANNEL_ID)
                .setSmallIcon(R.drawable.notification_icon_background)
                .setContentTitle(title)
                .setContentText(content)
                .setPriority(NotificationCompat.PRIORITY_DEFAULT);

        NotificationManager manager = (NotificationManager)getSystemService(Context.NOTIFICATION_SERVICE);
        manager.notify(channel_num, builder.build());

    }

    private void removeNotification(){
        NotificationManager manager = (NotificationManager)getSystemService(Context.NOTIFICATION_SERVICE);
        manager.cancelAll();
        channel_num = - 1;
    }

    private void createNoficationChannel(){
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O){
            CharSequence name = title;
            String description = content;
            int importance = NotificationManager.IMPORTANCE_DEFAULT;
            NotificationChannel channel = new NotificationChannel(CHANNEL_ID, name, importance);
            channel.setDescription(description);
            NotificationManager notificationManager = getSystemService(NotificationManager.class);
            notificationManager.createNotificationChannel(channel);

        }
    }


}

File: 4-1/임베실/week13/app/src/main/java/com/example/week13/MainActivity.kt
package com.example.week13

import android.annotation.SuppressLint
import android.app.NotificationChannel
import android.app.NotificationManager
import android.content.Context
import android.os.Build
import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.RadioButton
import android.widget.Toast
import androidx.annotation.RequiresApi
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.NotificationCompat
import java.lang.String

class MainActivity : AppCompatActivity() {
    var text : EditText? = null
    var CHANNEL_ID = "channel_ID"
    var inputValue : Float? = null
    var outputValue : Float? = null

    @RequiresApi(Build.VERSION_CODES.O)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        text = findViewById(R.id.inputValue)

        createNoficationChannel()

    }
    fun onClick(view: View) {
        when (view.getId()) {
            R.id.button1 -> {
                val celsiusButton : RadioButton = findViewById(R.id.radio0)
                val fahrenheitButton : RadioButton = findViewById(R.id.radio1)
                if (text!!.text.length == 0) {
                    Toast.makeText(
                        this, "Please enter a valid number",
                        Toast.LENGTH_LONG
                    ).show()
                    return
                }
                inputValue = text!!.text.toString().toFloat()
                if (celsiusButton.isChecked) {
                    outputValue = (inputValue!! -32) * 5 / 9
                    text!!.setText(
                        String
                            .valueOf(outputValue!!)
                    )
                    celsiusButton.isChecked = false
                    fahrenheitButton.isChecked = true
                    addNotification1()
                } else {
                    outputValue = (inputValue!! * 9 ) / 5 + 32
                    text!!.setText(
                        String
                            .valueOf(outputValue!!)
                    )
                    fahrenheitButton.isChecked = false
                    celsiusButton.isChecked = true
                    addNotification2()
                }
            }
        }
    }

    @SuppressLint("PrivateResource")
    private fun addNotification1() {
        var builder =
            NotificationCompat.Builder(this, CHANNEL_ID)
                .setSmallIcon(R.drawable.notification_icon_background)
                .setContentTitle("week13")
                .setContentText("$inputValue fahrenheit is converted to $outputValue celcius")
                .setPriority(NotificationCompat.PRIORITY_DEFAULT)
        var manager =
            getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
        manager.notify(0, builder.build())
    }

    @SuppressLint("PrivateResource")
    private fun addNotification2() {
        var builder =
            NotificationCompat.Builder(this, CHANNEL_ID)
                .setSmallIcon(R.drawable.notification_icon_background)
                .setContentTitle("week13")
                .setContentText("$inputValue celcius is converted to $outputValue fahrenheit")
                .setPriority(NotificationCompat.PRIORITY_DEFAULT)
        var manager =
            getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
        manager.notify(0, builder.build())
    }


    @SuppressLint("PrivateResource")
    private fun createNoficationChannel() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            val name = "week13"
            val descriptionText = "week13"
            val importance = NotificationManager.IMPORTANCE_DEFAULT
            val channel = NotificationChannel(CHANNEL_ID, name, importance)
            channel.setDescription(descriptionText)
            val notificationManager: NotificationManager =
                getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
            notificationManager.createNotificationChannel(channel)
        }
    }


}


